### DISCUSSION TITLE
After installing using HELM in EKS, PostgreSQL consistently crashes LoopBackOff

### DISCUSSION THREAD

**Discussion by kaneenak** [apache/airflow#35384] [General]
### Official Helm Chart version

1.10.0

### Apache Airflow version

2.6.2

### Kubernetes Version

1.28.2

### Helm Chart configuration

airflow:
  legacyCommands: false
  image:
    repository: apache/airflow
    tag: 2.6.2

  executor: KubernetesExecutor

  config:
    AIRFLOW__WEBSERVER__EXPOSE_CONFIG: "False"
    AIRFLOW__CORE__LOAD_EXAMPLES: "False"

  variables: []

  pools: []

  extraPipPackages: []

scheduler:
  replicas: 1

web:
  replicas: 1
  service:
    type: ClusterIP
    externalPort: 8080

workers:
  replicas: 1
  persistence:
    enabled: false
    storageClassName: efs-sc
    size: 10Gi

flower:
  enabled: false

logs:
  persistence:
    enabled: false

dags:    
  gitSync:
    enabled: true
    repo: https://git-codecommit.cn-northwest-1.amazonaws.com.cn/v1/repos/****
    branch: airflow
    credentialsSecret: airflow-credentials-secret

extraSecrets:
  airflow-credentials-secret:
    data: |
      GIT_SYNC_USERNAME:***********
      GIT_SYNC_PASSWORD: *******************

ingress:
  enabled: false

postgresql:
  enabled: true
  persistence:
    enabled: true
    storageClass: efs-sc
    size: 8Gi

redis:
  enabled: True
  persistence:
    enabled: true
    storageClassName: efs-sc
    size: 8Gi

triggerer:
  enabled: true
  replicas: 1
  persistence:
    enabled: true
    storageClassName: efs-sc
    size: 10Gi

### What happened

I used helm to install airflow in AWS EKS. After executing the install command, the postgreSQL pod remained in a CrashLoopBackOff state. I used the kubectl logs command to check the pod log and found no abnormalities. The detailed information is as follows
![image](https://github.com/apache/airflow/assets/122849336/1278d14e-086b-429a-a69a-9af48e6f5510)

PostgreSQL was unable to start properly, causing my other pods' wait for air flow migrations (init) container to fail. His log error was that the postgreSQL service could not be queried, which resulted in most of my pods being in an initialization state
![image](https://github.com/apache/airflow/assets/122849336/674fbb07-d241-4db5-8486-3e5e05a1a00c)

### What you think should happen instead

I don't know why it caused the pods of PostgreSQL to continue to CrashLoopBackOff. The persistent storage I configured for him uses AWS's EFS

### How to reproduce

Configure EFS persistent storage using AWS eks 1.28 version, and then execute install with my value.yaml file to reproduce

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by kaneenak** [comment]
I hope you can help me solve this problem

**Comment by jscheffl** [comment]
Postgres within the helm deployment is for testing and development use only. It is recommended to use a "real" postgres, probably as a PaaS offering if you don't want to run your own, if you want to run Airflow.

Otherwise if you insist on running Postgres within docker, there might be multiple reasons. One is that the persistent storage has no writer permissions, ressources are too low. Please increase log level of the postgres docker and inspect logs of the docker, Especially try to follow logs with `kubectl logs -f ...` to fetch the last messages before crash.

**Comment by potiuk** [comment]
Converting to a discussion. Please avoid opening "airflow isssues" when you ask a troubleshooting question - it's clearly explained in the issue template that you should open discussion if you want to ask for it not an issue with link to discussions @kaneenak

**Comment by thatsdone** [comment]
Hi, 
I got a similar symptom recently.

In case of me, I was using in-house kubernetes cluster using 1.30, and the root cause was there was no valid default storageClass.
After specifying values.yml like below, I could successfully deploy Airflow (for evaluation purpose).

I think the original case was the same root cause because on EKS there is no usable storage class unless users deployed EBS (or EFS) csi plugin.

```yaml
dags:
  persistence:
    storageClassName: myStorageClass
logs:
  persistence:
    storageClassName: myStorageClass
workers:
  persistence:
    storageClassName: myStorageClass
triggerer:
  persistence:
    storageClassName: myStorageClass
redis:
  persistence:
    storageClassName: myStorageClass
# BELOW is important and not documented.
postgresql: 
  global:
    storageClass: myStorageClass
```

Here, what I noticed is like below.
* PostgreSQL is deployed using bitnami/postgresql subchart
    * https://artifacthub.io/packages/helm/bitnami/postgresql 
* Granted that postgresql deployment is done by an external chart, but still Airflow helm chart documentation
does not have enough description
  * Note that dags/logs/workers/triggerer and even redis (!) have explicit explanation, but not for postgresql.
* It's better to write explicitly in the documentation (https://airflow.apache.org/docs/helm-chart/stable/parameters-ref.html, not only in chart/RELEASE_NOTES.rst) that: 
    * all the variables under 'postgres' key will be passed to the subchart
    * Thus, `postgres.A.B.C=VALUE` would be passed to the bitnami/postgresql subchart as `A.B.C=VALUE`

Hope this is useful for the community.
